### PR TITLE
A tiny-js-world-task

### PR DESCRIPTION
**Pull request by PAtrofimov**
### a-tiny-js-world
[Code base](https://github.com/PAtrofimov/a-tiny-JS-world)
[Demo](https://patrofimov.github.io/a-tiny-JS-world/)
The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+let addFriend = function(friend) {
+    if (this.friends.indexOf(friend) == -1) {
+        this.friends.push(friend);
+        friend.friends.push(this);
+    }
+}
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: Very nice solution to make friends mutual in a single step.

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+man.addFriend(woman);
+man.addFriend(dog);
+
+listObj = [];
+listObj.push(woman);
+listObj.push(man);
+listObj.push(cat);
+listObj.push(dog);
+listObj.push(womancat);
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: The above will read better if `let listObj = [ woman, man, ......];`
Please, also give the variable a name without abbreviations. Autocompletion in IDEs does the dirty job for us and future contributors will be grateful for self-explaining and consistent names.
(Obj can stand for Objection, Objective, and Object)

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+let objToString = function(obj, lim = ';') {
+    let rezult = '';
+    for (key in obj) {
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: Order of properties in an object is not guaranteed.
Please, list properties explicitly.
This function body can be simplified and also read better.
We write code for other people, firsthand.

Hint:
```
[
  varA,
  condition ? varB : varC,
  someArray.join(','),
].join(delimiter) + (anotherArray.length ? anotherArray.join(',') + ';' : '');
```

P.S. BTW, "lim" stands for "LIMit" or "deLIMiter"?

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        } else rezult += obj[key];
+        rezult += lim;
+
+    }
+    return rezult;
+}
+
+listObj.forEach(function(obj) {
+	if (obj.hands != 0)
+    print(objToString(obj));
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: Please, fix indentation. We write code for other people, firsthand.
You can tune your IDE to do the dirty job for you.
You may also employ es6 features like arrow functions.

**Comment by PAtrofimov** [comment]
Fixed everything.

**Comment by OleksiyRudenko** [comment]
@PAtrofimov thanks. How about demo?

**Comment by PAtrofimov** [comment]
I corrrect description and add demo.

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    inhabitant.legs,
+    inhabitant.hands,
+    inhabitant.saying,
+    inhabitant.friends.length ? inhabitant.friends.map((friend) =>friend.name).join(',') : ''    
+    ].join(delimiter);
+    return rezult;
+}
+
+inhabitants.forEach((inhabitant) => {
+    if (inhabitant.hands != 0) print(getDescription(inhabitant));
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: Cannot see any good reason to discriminate handless inhabitants.
Please, eliminate discrimination when you're back to your world after the following task.

File: submissions/patrofimov/a-tiny-js-world/index.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+});
```
Comment (OleksiyRudenko) [inline submissions/patrofimov/a-tiny-js-world/index.js]: Having an empty line at the file end is important. This protects your code from being marked as changed when the only change in a line is an newline character.
Please, tune your IDE to do this dirty job (ensure a single empty line at the file end) for you.

**Pull request merged**